src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -253,20 +253,13 @@ private PodTemplateCallback(PodTemplate podTemplate) {
          * Remove the template after step is done
          */
         protected void finished(StepContext context) throws Exception {
-            Cloud cloud = Jenkins.get().getCloud(cloudName);
-            if (cloud == null) {
-                LOGGER.log(Level.FINE, "Cloud {0} no longer exists, cannot delete pod template {1}",
-                        new Object[] { cloudName, podTemplate.getName() });
-                return;
-            }
-            if (cloud instanceof KubernetesCloud) {
-                LOGGER.log(Level.FINE, "Removing pod template {1} from cloud {0}",
-                        new Object[] { cloud.name, podTemplate.getName() });
-                KubernetesCloud kubernetesCloud = (KubernetesCloud) cloud;
-                kubernetesCloud.removeDynamicTemplate(podTemplate);
-            } else {
-                LOGGER.log(Level.WARNING, "Cloud is not a KubernetesCloud: {0} {1}",
-                        new String[] { cloud.name, cloud.getClass().getName() });
+            try {
+                KubernetesCloud cloud = resolveCloud();
+                LOGGER.log(Level.FINE, () -> "Removing pod template " + podTemplate.getName()
+                        + " from cloud " + cloud.name);
+                cloud.removeDynamicTemplate(podTemplate);
+            } catch (AbortException e) {
+                LOGGER.log(Level.WARNING, () -> "Unable to resolve cloud for " + podTemplate.getName() + ". Maybe the cloud was removed while running the build?");
```

Comment:
Is it a good idea to mute this exception?

Reply:
Already updated in be75374